chips/nrf52840/src/interrupt_service.rs

```diff
@@ -0,0 +1,24 @@
+use crate::peripheral_interrupts;
+use nrf52::interrupt_service::InterruptService;
+
+pub struct Nrf52840InterruptService {
+    nrf52: nrf52::interrupt_service::Nrf52InterruptService,
+}
+
+impl Nrf52840InterruptService {
+    pub unsafe fn new(gpio_port: &'static nrf52::gpio::Port) -> Nrf52840InterruptService {
+        Nrf52840InterruptService {
+            nrf52: nrf52::interrupt_service::Nrf52InterruptService::new(gpio_port),
+        }
+    }
+}
+
+impl InterruptService for Nrf52840InterruptService {
+    unsafe fn service_interrupt(&self, interrupt: u32) -> bool {
+        match interrupt {
+            peripheral_interrupts::USBD => nrf52::usbd::USBD.handle_interrupt(),
+            _ => return self.nrf52.service_interrupt(interrupt),
```

Comment:
Can we just handle the GPIO interrupt here and avoid the need to pass around the GPIO object?

Reply:
I think it adds unnecessary confusion to a reader who has to trace through why `gpio_port` is passed in when nothing else is, and when no other boards do it this way. If more peripherals end up needing chip-specific objects this gets even more complicated, particularly since this elegant InterruptService trait provides a very natural way to avoid it.

Reply:
I'm opposed to that. If someone implements a new 52833 chip for example, they may forget to handle the `GPIOTE` interrupt with your proposal. On the other hand, with the current code, they cannot avoid passing a suitable `gpio_port` object to the `Nrf52InterruptService::new` constructor. So this API prevents mistakes, besides reducing code duplication (i.e. having the same `GPIOTE => {...}` handler in all the chips - 52832, 52840, and whatever other chips there would be).

That said, there's not much reason for `Nrf52840InterruptService`'s new function to take a `gpio_port` as argument, given that it's always `nrf52840`'s `gpio::PORT`. I can change that.

Reply:
I agree with @gendx